docs/frameworks/sveltekit.md

```diff
@@ -0,0 +1,229 @@
+to disable it on your application.
+
+Since `SvelteKit` will use `SSR / SSG`, we need to add the `ReloadPrompt` component using `dynamic import`. `Vite Plugin PWA`
+will only register the service worker on build, it is aligned with the current behavior of
+[SvelteKit service worker module](https://kit.svelte.dev/docs#modules-$service-worker) <outbound-link />.
+
+The best place to include the `ReloadPrompt` is on the main layout of the application:
+
+<details>
+  <summary><strong>src/routes/__layout.svelte</strong> code</summary>
+
+```html
+<script>
+  import { onMount } from 'svelte'
+  import { browser, dev } from '$app/env'
+
+  let ReloadPrompt
+  onMount(async () => {
+    !dev && browser && (ReloadPrompt = (await import('$lib/components/ReloadPrompt.svelte')).default)
+  })
+</script>
+
+<svelte:head>
+  {#if (!dev && browser)}
+  <link rel="manifest" href="/_app/manifest.webmanifest">
+  {/if}
+</svelte:head>
+
+<main>
+  <slot />
+</main>
+
+{#if ReloadPrompt}
+<svelte:component this={ReloadPrompt} />
+{/if}
+```
+</details>
+
+## SvelteKit Adapters
+
+The main problem with the current implementation of the service worker module of `SvelteKit` is that you don't have access
+to the result applied by any adapter you have configured on your application.
+
+The service worker module of `SvelteKit` will be called before the adapter logic is applied, and so, inside the service worker
+module, you don't have access to those resources.
+
+Your application will not work when the user is offline, since the pages will not be included on the service worker precache
+manifest.
+
+When using `Vite PWA Plugin` with any `SvelteKit Adapter` you need to provide an additional script to rebuild your `pwa`
+once `SvelteKit` finish building your application, that is, when the adapter configured finish its job.
+
+The biggest difference between this plugin and the SvelteKit service worker module is that this plugin does not require
+integration into the application logic, just configuration.
+
+You can take a look at [SvelteKit example](https://github.com/antfu/vite-plugin-pwa/tree/main/examples/sveltekit-pwa) <outbound-link />
+to configure the additional scripts on your application, it is quite complex since we use it for multiple behaviors 
+with the same codebase.
+
+As an example using [@sveltejs/adapter-static](https://github.com/sveltejs/kit/tree/master/packages/adapter-static) <outbound-link />
+with `generateSW` strategy and `Prompt for update` behavior, you will need:
+
+<details>
+<summary>1) add <strong>pwa.js</strong> script</summary>
+
+```js
+import { resolveConfig } from 'vite'
+import { VitePWA } from 'vite-plugin-pwa';
+import { pwaConfiguration } from './pwa-configuration.js';
+import { copyFileSync } from 'fs';
+
+const webmanifestDestinations = [
+	'./.svelte-kit/output/client/',
+	'./build/',
+]
+
+const swDestinations = [
+	'./build/',
+]
+
+const buildPwa = async() => {
```

Comment:
it seems like it'd be nice to do this in the Vite plugin. what does it need access to that prevents that from being possible? I wonder if https://github.com/sveltejs/kit/pull/2580 would be enough? Or would you also need to do things like compress assets in a Vite plugin as well?

Reply:
Here we have a few problems:
1) base path used by SvelteKit is `/_app/` and not `/`: just comment out `base` and `scope` from the pwa configuration and build the example project
2) previous statement also applies for the service worker manifest `scope` and `base` entries
3) the `api` exposed by this plugin is for this purpose: we have integrated on `vite-ssg`, proposed to be included on `vitepress` but rejected by Evan and I have some branch on my SvelteKit repo with the integration